### PR TITLE
Benchmark `Email` implementations

### LINKED ISSUE #9: Benchmark `Email` against other libraries
Could be fun to compare.

### PR DESCRIPTION
**Pull request by meduzen**
## Checklist

- [x] I’m aware of the [contributing guidelines](https://github.com/frontacles/frontacles/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/frontacles/frontacles/blob/main/CODE_OF_CONDUCT.md).
- [ ] The code I add comes with:
  - [ ] tests
  - [ ] types (`npm run types` to generate them from JSDoc comments)
  - [x] [documentation](https://github.com/frontacles/frontacles/blob/main/README.md)
  - [x] [changelog](https://github.com/frontacles/frontacles/blob/main/CHANGELOG.md)
  - [ ] [`size-limit` config](https://github.com/frontacles/frontacles/blob/main/.size-limit.json)
  - [x] [benchmarks](https://github.com/frontacles/frontacles/blob/main/CONTRIBUTING.md#benchmarks)

---

Closes #9.

### CODE REVIEW

File: src/url/bench/email.results.txt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+✓ src/url/bench/email.bench.js > url/email passing all tests 9054ms
+     name                                               hz     min     max    mean     p75     p99    p995    p999     rme  samples
+   · `EmailWithURL.canParse` (current)        2,179,868.71  0.0003  1.6480  0.0005  0.0005  0.0007  0.0008  0.0024  ±0.83%  1089935
+   · `EmailClassWithRegExp.canParse`         12,644,349.22  0.0000  0.1205  0.0001  0.0001  0.0001  0.0001  0.0003  ±0.07%  6322175   fastest
+   · `emailWithRegExp`                       12,591,750.06  0.0000  0.0951  0.0001  0.0001  0.0001  0.0001  0.0003  ±0.09%  6295876
+   · `validator/isEmail`                        687,041.51  0.0013  1.0840  0.0015  0.0015  0.0018  0.0018  0.0023  ±0.57%   343521
+   · yup create schema and `isValidSync`        583,325.45  0.0015  1.0215  0.0017  0.0017  0.0023  0.0025  0.0039  ±0.65%   291663   slowest
+   · yup precalculated schema `isValidSync`   2,119,568.48  0.0004  0.8205  0.0005  0.0005  0.0009  0.0010  0.0014  ±0.35%  1059785
+
+ ✓ src/url/bench/email.bench.js > url/email not passing all tests 6586ms
```
Comment (meduzen) [inline src/url/bench/email.results.txt:10]: All libraries in this category (joi, Valibot, Zod) are only failing on edge cases.

**Comment by meduzen** [comment]
Conclusions from this PR:

- `Email.canParse` is as fast or faster than libraries **passing all tests**.
- When re-using precalculated validation schemas, `valibot` and `zod` are up to 400% faster, but they **fail** at some edge cases tests. When not reusing schemas (one-shot validation), they are up to 400% slower.
- Almost all libraries use RegExp for validation. Reading their implementation, I stumbled on the [WHATWG RegeExp](https://html.spec.whatwg.org/multipage/input.html#valid-e-mail-address) and added some implementations to use it, which is 150% faster than precalculated `zod` and 600% faster than Frontacles `Emain.canParse`. Definitely a great finding.
- See #36 (ongoing) for decisions regarding email validation in Frontacles.

File: src/url/bench/email.results.txt (hunk 1 above)
Comment (meduzen) [reply src/url/bench/email.results.txt]: Elaborating in https://github.com/frontacles/frontacles/pull/35#issuecomment-2708188805.
